### PR TITLE
Enable HAPROXY for Hadoop HTTPFS access

### PR DESCRIPTION
**Pull request by amithkanand**
This PR moves `Hadoop HTTPFS` protocol behind the cluster `VIP` address. 

To access `HTTPFS` using VIP on a Kerberized cluster use:

```
$ curl  --location-trusted --negotiate -u ubuntu "http://192.168.100.5:14000/webhdfs/v1?op=liststatus"
```

On a non-kerberized cluster:

```
$ curl  --location "http://192.168.100.5:14000/webhdfs/v1?op=liststatus"
```

### CODE REVIEW

File: cookbooks/bcpc/recipes/haproxy.rb [hunk 1]
```diff
@@ -55,8 +55,10 @@
 template "/etc/haproxy/haproxy.cfg" do
   source "haproxy.cfg.erb"
   mode 00644
-  variables(:mysql_servers => get_nodes_for("mysql","bcpc")
-  notifies :restart, "service[haproxy]", :immediately
+  variables(:mysql_servers => get_nodes_for("mysql","bcpc"),
+   :httpfs_servers => get_nodes_for("datanode","bcpc-hadoop")
```
Comment (cbaenziger) [inline cookbooks/bcpc/recipes/haproxy.rb]: Just a nit: one space instead of two here...

File: cookbooks/bcpc/templates/default/haproxy.cfg.erb [hunk 2]
```diff
@@ -65,3 +65,14 @@ option tcpka
   <%= "server #{server['hostname']} #{server['bcpc']['management']['ip']}:3306 check port 3307 inter 1s rise 1 fall 1 observe layer4 backup" %>
 <% end -%>
 <% end -%>
+
+<% if @httpfs_servers.length > 0 -%>
+listen httpfs <%="#{node[:bcpc][:floating][:vip]}"%>:14000
+  timeout  client 24h
+  timeout  server 24h
```
Comment (cbaenziger) [inline cookbooks/bcpc/templates/default/haproxy.cfg.erb]: Erm, two spaces after the `timeout`'s

File: cookbooks/bcpc/templates/default/haproxy.cfg.erb [hunk 3]
```diff
@@ -65,3 +65,14 @@ option tcpka
...
 <% end -%>
+
+<% if @httpfs_servers.length > 0 -%>
+listen httpfs <%="#{node[:bcpc][:floating][:vip]}"%>:14000
+  timeout client 24h
+  timeout server 24h
+  balance leastconn
+  option httpchk GET / 
+<% @httpfs_servers.each do |server| -%>
+  <%= "redirect code 301 prefix http://#{float_host(server['fqdn'])}:14000" %>
```
Comment (cbaenziger) [inline cookbooks/bcpc/templates/default/haproxy.cfg.erb]: We should have some [health checking](http://www.haproxy.com/doc/aloha/7.0/haproxy/healthchecks.html) defined here

**Comment by vt0r** [comment]
Closing for cleanup. Feel free to reopen if this work gets picked up later on.

**Pull request closed without merging**